src/status_im/stickers/core.cljs

```diff
@@ -17,24 +28,102 @@
+
+(defn prepare-transaction [id tx on-result]
+  (merge {:id     id
+          :symbol :ETH
+          :method constants/web3-send-transaction
+          :amount (money/bignumber 0)}
+         (when on-result {:on-result on-result})
+         tx))
+
+(fx/defn buy-token [{{:keys [network] :as db} :db} pack-id]
+  (let [network   (get-in db [:account/account :networks network])
+        address   (ethereum/normalized-address (get-in db [:account/account :address]))
+        tx-object {:to   (get ethereum.stickers/contracts (ethereum/network->chain-keyword network))
+                   :data (abi-spec/encode "buyToken(uint256,address)" [pack-id address])}]
+    (models.wallet/open-modal-wallet-for-transaction
+     db
+     (prepare-transaction "buy" tx-object [:stickers/pending-pack pack-id])
+     tx-object)))
+
+(fx/defn approve-pack [{db :db} pack-id price]
+  (let [network           (get-in db [:account/account :networks (:network db)])
+        chain             (ethereum/network->chain-keyword network)
+        stickers-contract (get ethereum.stickers/contracts chain)
+        tx-object         {:to   (get erc20/snt-contracts chain)
+                           :data (abi-spec/encode "approve(address,uint256)" [stickers-contract price])}]
+    (models.wallet/open-modal-wallet-for-transaction
+     db
+     (prepare-transaction "approve" tx-object [:stickers/buy-token pack-id])
+     tx-object)))
+
+(fx/defn pending-pack [{{:keys [web3 network] :as db} :db :as cofx} id]
+  (let [network (get-in db [:account/account :networks network])
+        address (ethereum/normalized-address (get-in db [:account/account :address]))]
+    (fx/merge cofx
+              {:db (update db :stickers/packs-pendning conj id)
+               :stickers/owned-packs-fx [web3 network address]}
+              (navigation/navigate-to-clean :wallet-transaction-sent-modal {})
+              #(when (zero? (count (:stickers/packs-pendning db)))
+                 {:stickers/set-pending-timout-fx nil}))))
+
+(fx/defn pending-timeout [{{:keys [web3 network] :stickers/keys [packs-pendning packs-owned] :as db} :db}]
+  (let [packs-diff (clojure.set/difference packs-pendning packs-owned)
+        network (get-in db [:account/account :networks network])
+        address (ethereum/normalized-address (get-in db [:account/account :address]))]
+    (merge {:db (assoc db :stickers/packs-pendning packs-diff)}
+           (when-not (zero? (count packs-diff))
+             {:stickers/owned-packs-fx [web3 network address]
+              :stickers/set-pending-timout-fx nil}))))
+
+(re-frame/reg-fx
+ :stickers/pack-data-fx
+ (fn [[web3 network id open?]]
+   (when-let [contract (get ethereum.stickers/contracts (ethereum/network->chain-keyword network))]
+     (ethereum.stickers/pack-data web3 contract id (pack-data-callback id open?)))))
+
+(re-frame/reg-fx
+ :stickers/set-pending-timout-fx
+ (fn []
+   (js/setTimeout #(re-frame/dispatch [:stickers/pending-timout]) 10000)))
```

Comment:
Don't we have a generic effect for this?

Reply:
haven't seen, i'll take a look thanks

Reply:
we do in utils there is already a set-timeout